src/test/java/com/intel/gkl/compression/DeflaterUnitTest.java

```diff
@@ -81,14 +81,54 @@ public void randomDNATest() {
             final IntelInflater inflater = new IntelInflater(true);
 
                 randomDNA(input);
+
+                log.debug("Negative testing for offset.");
+                try {
+                    deflater.setInput(input, -1, input.length);
+                    Assert.fail("IllegalArgumentException expected");
+                }
+                catch(IllegalArgumentException e) {}
+
+                try {
+                    deflater.setInput(input, input.length, input.length);
+                    Assert.fail("IllegalArgumentException expected");
+                }
+                catch(IllegalArgumentException e) {}
+
+                try {
+                    deflater.setInput(input, input.length + 1, input.length);
+                    Assert.fail("IllegalArgumentException expected");
+                }
+                catch(IllegalArgumentException e) {}
+
                 deflater.setInput(input, 0, input.length);
                 deflater.finish();
 
                 int compressedBytes = 0;
+
                 while (!deflater.finished()) {
                     compressedBytes = deflater.deflate(compressed, 0, compressed.length);
                 }
 
+                log.debug("Deflate parameters negative testing.");
+                try {
+                    deflater.deflate(null, 0, compressed.length);
+                    Assert.fail("NullPointerException expected.");
+                }
+                catch(NullPointerException e) {}
```

Comment:
@mkuligow - I think adding negative tests from functinal `randomDNA` test, will help in  isolating the different scenarios